### DISCUSSION TITLE
METV11.1.0 Grid-Stat appears to have been compiled with OpenMP?

### DISCUSSION THREAD

**Discussion by jwraby** [dtcenter/METplus#2328] [Existing Builds] [answered]
I ran Grid-Stat in the MET container on the HPC in an interactive batch session. For some reason, the following DEBUG statement was output in the screen dialog:
Error:DEBUG 2: OMP_NUM_THREADS is not set. Defaulting to 1 thread. Recommend setting OMP_NUM_THREADS for faster runtimes.

The HPC Help Desk informed that I needed to set the environment variable OMP_NUM_THREADS to a value from 1 to 96. The 96 value comes from the fact that each node is capable of 96 threads. They made the statement that "The code is not serial if it's expecting that variable to be set." My understanding is that the MET code is serial. To test this I requested 1 thread per node instead of 96 and ran Grid-Stat and it ran in about 90 secs. I then requested 96 threads and re-ran Grid-Stat but it ran about 90 secs too. I conclude that the MET code is not parallel and must be serial or the run with 96 threads would have executed much faster. Was the MET code compiled using OpenMP? Thanks.

**Comment by jprestop** [comment, marked as answer]
HI @jwraby.  The HPC Help Desk is correct that the environment variable OMP_NUM_THREADS needs to be set. Please take a look at this [OMP_NUM_THREADS](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#omp-num-threads) documentation.  We recently added some parallelization that's currently _only_ utilized in the computation of neighborhood methods in Grid-Stat. We do plan to extend our use of Open MP in the future.

**Comment by jwraby** [comment]
Thanks for this information, Julie. Good to know. 

R/

John

 

From: jprestop ***@***.***> 
Sent: Thursday, August 31, 2023 10:25 AM
To: dtcenter/METplus ***@***.***>
Cc: Raby, John W CIV USARMY DEVCOM ARL (USA) ***@***.***>;
Mention ***@***.***>
Subject: [Non-DoD Source] Re: [dtcenter/METplus] METV11.1.0 Grid-Stat
appears to have been compiled with OpenMP? (Discussion #2328)

 

HI @jwraby
<https://usg01.safelinks.protection.office365.us/?url=https%3A%2F%2Fgithub.c
om%2Fjwraby&data=05%7C01%7Cjohn.w.raby2.civ%40army.mil%7C1452bf2ec4d24bad5f7
c08dbaa3ee4e0%7Cfae6d70f954b481192b60530d6f84c43%7C0%7C0%7C63829095933941064
1%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haW
wiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=GO6TLzsvqouuAwNurWFfcEf%2BS3rh402jy8N
Edzu4pnM%3D&reserved=0> . The HPC Help Desk is correct that the environment
variable OMP_NUM_THREADS needs to be set. Please take a look at this
OMP_NUM_THREADS
<https://usg01.safelinks.protection.office365.us/?url=https%3A%2F%2Fmet.read
thedocs.io%2Fen%2Flatest%2FUsers_Guide%2Fconfig_options.html%23omp-num-threa
ds&data=05%7C01%7Cjohn.w.raby2.civ%40army.mil%7C1452bf2ec4d24bad5f7c08dbaa3e
e4e0%7Cfae6d70f954b481192b60530d6f84c43%7C0%7C0%7C638290959339410641%7CUnkno
wn%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI
6Mn0%3D%7C3000%7C%7C%7C&sdata=%2FsnY2yWRmsNmA0%2FYj%2BYm8vdkiojE2PlhLvp%2F0U
OlH4c%3D&reserved=0>  documentation. We recently added some parallelization
that's currently only utilized in the computation of neighborhood methods in
Grid-Stat. We do plan to extend our use of Open MP in the future.

-
Reply to this email directly, view it on GitHub
<https://usg01.safelinks.protection.office365.us/?url=https%3A%2F%2Fgithub.c
om%2Fdtcenter%2FMETplus%2Fdiscussions%2F2328%23discussioncomment-6877718&dat
a=05%7C01%7Cjohn.w.raby2.civ%40army.mil%7C1452bf2ec4d24bad5f7c08dbaa3ee4e0%7
Cfae6d70f954b481192b60530d6f84c43%7C0%7C0%7C638290959339410641%7CUnknown%7CT
WFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3
D%7C3000%7C%7C%7C&sdata=f%2B9G79aHIZdToPjOyPtxiejs2gwZryNhVJQJ4gzespM%3D&res
erved=0> , or unsubscribe
<https://usg01.safelinks.protection.office365.us/?url=https%3A%2F%2Fgithub.c
om%2Fnotifications%2Funsubscribe-auth%2FAUX5FH53HCMSZQBFVIMTJALXYC3ELANCNFSM
6AAAAAA4DQ47NA&data=05%7C01%7Cjohn.w.raby2.civ%40army.mil%7C1452bf2ec4d24bad
5f7c08dbaa3ee4e0%7Cfae6d70f954b481192b60530d6f84c43%7C0%7C0%7C63829095933941
0641%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1
haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=B609x17Qb0T1PllDkxyUSDWDsZT%2FJjSu
ucNuA4%2B8zyg%3D&reserved=0> .
You are receiving this because you were mentioned.
<https://github.com/notifications/beacon/AUX5FH7ZZWROS6XA6CLXBKLXYC3ELA5CNFS
M6AAAAAA4DQ47NCWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3T
UL5UWJTQANDZBM.gif> Message ID:
***@***.***>

**Reply by jprestop** [reply]
Thanks @jwraby.  Do you have any more questions or can we mark this Discussion as answered?

**Comment by jwraby** [comment]
Julie - 

 

I have no more questions. Please mark as answered for me. 

 

Thanks.

 

R/

John

 

From: jprestop ***@***.***> 
Sent: Wednesday, September 6, 2023 11:37 AM
To: dtcenter/METplus ***@***.***>
Cc: Raby, John W CIV USARMY DEVCOM ARL (USA) ***@***.***>;
Mention ***@***.***>
Subject: [Non-DoD Source] Re: [dtcenter/METplus] METV11.1.0 Grid-Stat
appears to have been compiled with OpenMP? (Discussion #2328)

 

Thanks @jwraby
<https://usg01.safelinks.protection.office365.us/?url=https%3A%2F%2Fgithub.c
om%2Fjwraby&data=05%7C01%7Cjohn.w.raby2.civ%40army.mil%7Ca8c23fe75a4043579b8
e08dbaefffef1%7Cfae6d70f954b481192b60530d6f84c43%7C0%7C0%7C63829618677220133
3%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haW
wiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=40FFuOlAnFU9jfKussHJwfQJXWJDUePHDdnaZ
yfUbWE%3D&reserved=0> . Do you have any more questions or can we mark this
Discussion as answered?

-
Reply to this email directly, view it on GitHub
<https://usg01.safelinks.protection.office365.us/?url=https%3A%2F%2Fgithub.c
om%2Fdtcenter%2FMETplus%2Fdiscussions%2F2328%23discussioncomment-6928649&dat
a=05%7C01%7Cjohn.w.raby2.civ%40army.mil%7Ca8c23fe75a4043579b8e08dbaefffef1%7
Cfae6d70f954b481192b60530d6f84c43%7C0%7C0%7C638296186772201333%7CUnknown%7CT
WFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3
D%7C3000%7C%7C%7C&sdata=ZI0Wah7ZaFsAQu0YQBRG17CSX0drBTe%2F3%2BaVLA%2BugDA%3D
&reserved=0> , or unsubscribe
<https://usg01.safelinks.protection.office365.us/?url=https%3A%2F%2Fgithub.c
om%2Fnotifications%2Funsubscribe-auth%2FAUX5FH3KNXU2G3USXJX5JELXZCYEBANCNFSM
6AAAAAA4DQ47NA&data=05%7C01%7Cjohn.w.raby2.civ%40army.mil%7Ca8c23fe75a404357
9b8e08dbaefffef1%7Cfae6d70f954b481192b60530d6f84c43%7C0%7C0%7C63829618677235
7550%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1
haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=nrtPMov63pqFdWUc8%2BBUIAUiSGmaeTXO
TVbVrkiepUU%3D&reserved=0> .
You are receiving this because you were mentioned.
<https://github.com/notifications/beacon/AUX5FH7G3LG5HDPIQPQXUQTXZCYEBA5CNFS
M6AAAAAA4DQ47NCWGG33NNVSW45C7OR4XAZNRIRUXGY3VONZWS33OINXW23LFNZ2KUY3PNVWWK3T
UL5UWJTQANG4QS.gif> Message ID:
***@***.***
***@***.***>
>
